### PR TITLE
PWADs can define a start map for skill selection

### PR DESCRIPTION
**Pull request by derekmd**
Fixes https://github.com/team-eternity/eternity/issues/417

Reintroduce Quake-style start maps. Doom, Doom II, or Heretic add-on WADs can define an `EDFROOT` lump (or base .edf file in a .pke) with gameproperties `menu.startmap "START"`. It will skip the skill & episode selection menus to allow an interactive level & `EMAPINFO` to define such behavior.

## Changes

1. Replace the unused CVAR `use_startmap` with a new EDF property.
    * I tried keeping the CVAR and allow PWADs to set it using `OPTIONS` line `use_startmap 1` but it ends up this WAD lump isn't read until a level is first loaded. So it isn't possible to alter "New Game" menu behavior using a flag defined in `OPTIONS`.
   * The CVAR was introduced by SMMU ([only when Doom II's IWAD was loaded](https://github.com/fragglet/smmu/blob/master/mn_menus.c#L125-L136)) to visit the map `START` on:
     * menu item "New Game"
     * console command `mn_newgame`
   * Eternity Engine disabled this feature after FraggleScript was removed. The included SMMU start.wad had to be converted to Small/ACS scripting which I think had feature gaps at the time?
2. Remove the unused commented-out code for the old SMMU menu item "Use the start map".
   * Without a `START` map always available in EE, this is not required.
   * Individual PWADs (hi Mordeth) will be enabling this feature via `EDFROOT`.
3. Remove start.wad from the `base/doom/eternity.pke` archive
   * To re-enable this feature for Doom, Doom II, & Heretic, that always-available `START` map's file must be removed
   * (... I designed this map in 1999 although it was really just a FraggleScript demo.)
   * Since Git won't show a ZIP file diff and you may not want rando contributors updating this file, I can remove this commit if you want a core team member to securely do it.

## Demo level

I significantly upgraded ye olde start.wad to showcase Eternity Engine's linked portals, polyobjects, and slopes: [dmdeestart.wad](http://derekmd.com/wp-content/uploads/2020/11/dmdeestart.zip)

Using a binary built from this branch, `eternity.exe -iwad doom2.wad -file dmdeestart.wad` and selected "New Game" will show the new EDF property is working.

dmdeestart-intro

## PWAD configuration:

**EDFROOT**

```
stdinclude("root.edf")

gameproperties
{
  menu.startmap "START"
}
```

**EMAPINFO** lump to skip the intermission screen on START map exit:
```
[START]
levelname = select your skill level
music = romero
skyname = SKY1
killstats = true
```

**START** level will trigger this ACS script to visit MAP01 on UV:
```
script 46 (void)
{
    ChangeSkill(SKILL_HARD);
    Teleport_NewMap(1, 0, 0);
}
```

> This pull request doesn't cover resetting the player's health & inventory when exiting to `MAP01`, as SMMU's start map feature was implemented to do since it essentially started a new Doom II game. Missing ZDoom ACS function `ChangeLevel()` will have to be added to cover this.

### CODE REVIEW

**Comment by derekmd** [comment]
... so it ends up the `OPTIONS` lump isn't loaded until the first level is started. see: `G_InitNew()`

So something else must be used by PWADs to enable `use_startmap`.

File: source/mn_menus.cpp [hunk 1]
```diff
@@ -264,12 +263,16 @@ CONSOLE_COMMAND(mn_newgame, 0)
 
    // haleyjd 05/14/06: check for episode menu override now
    if(mn_episode_override)
-   {
       MN_StartMenu(mn_episode_override);
-      return;
+   else if(GameModeInfo->menuStartMap &&
+           *GameModeInfo->menuStartMap &&
+           W_CheckNumForName(GameModeInfo->menuStartMap) >= 0)
```
Comment (derekmd) [inline source/mn_menus.cpp:269]: Unsure how to format this expression given the code style I've come across so far. Is there a code fixer / linter in the project?

Really would rather it be a class predicate but... Doom source pain.

```c++
  public:
      bool hasMenuStartMap()
      {
          return menuStartMap && *menuStartMap && W_CheckNumForName(menuStartMap) >= 0;
      }
```

Comment (Altazimuth) [reply source/mn_menus.cpp:269]: That else if is in keeping with the general style of EE. Sometimes I dunno how I feel about it, but it's sticking around for now at least.

**Comment by Altazimuth** [comment]
It was about time I merged this. It looks fine. There's a tiny thing I'll fix that's literally just a single character shift. Outside of that this all appears perfectly sound.

**Pull request merged**